tests/subscriptions/test_executor_consumer.py

```diff
@@ -356,6 +356,11 @@ def test_produce_result() -> None:
     strategy.poll()
     assert commit.call_count == 1
 
+    # Commit is throttled so if we immediately submit another message, the commit count will not change
+    strategy.submit(message)
+    strategy.poll()
+    assert commit.call_count == 1
```

Comment:
how do we know that the commit count didn't change because the future wasn't done?

Reply:
The throttled commit function is being once every time we poll regardless of whether the futures are done.

Reply:
ah yeah okies `self.__commit_data` could be `{}` but we still call `self.__commit`